components/collections/Cargo.toml

```diff
@@ -36,7 +36,7 @@ serde = { version = "1.0", default-features = false, features = ["derive", "allo
 databake = { version = "0.1.3", path = "../../utils/databake", features = ["derive"], optional = true }
 
 [dev-dependencies]
-postcard = { version = "1.0.0", features = ["alloc"] }
+postcard = { version = "1.0.0", features = ["alloc"], default-features = false }
```

Comment:
Nit/Question/Issue/Discussion: You had another PR a few weeks ago where you made similar changes in dev-dependencies, and I thought the conclusion was that we want to avoid going on a diet with dev-dependencies.

Reply:
Hm, fair. But it's just more annoying to diagnose these issues when dev-deps are involved. We keep getting `heapless` deps sneaking in and the dev-deps do not make it easier.

I think this is a low-risk low-reward change, not something I think we should expend effort in maintaining, but it's worth landing.

Reply:
Basically: I am not proposing we go on a diet. But these are changes I made anyway to in the process of diagnosing, and I'd like to include them because they're an improvement even if it's ephemeral.

Were I proposing we go on a diet, I would have added a depcheck test :smile: